src/test/regress/sql/view_propagation.sql

```diff
@@ -374,6 +374,68 @@ BEGIN;
     ALTER TABLE view_prop_schema_inner.view_in_transaction RENAME COLUMN a TO b;
     DROP VIEW view_prop_schema_inner.view_in_transaction;
 ROLLBACK;
+-- verify that the views get distributed after the table is distributed
+create table table_to_depend_on_1 (a int);
+create table table_to_depend_on_2 (a int);
+-- the first view depends on a table
+create view dependent_view_1 as select count(*) from table_to_depend_on_1;
+-- the second view depends on two tables
+create view dependent_view_2 as select count(*) from table_to_depend_on_1 join table_to_depend_on_2 on table_to_depend_on_1.a=table_to_depend_on_2.a;
+-- the third view depends on the first view
+create view dependent_view_3 as select count(*) from table_to_depend_on_1;
+-- the fourth view depends on the second view
+create view dependent_view_4 as select count(*) from table_to_depend_on_2;
+-- distribute only one table
+select create_distributed_table('table_to_depend_on_1','a');
+
+-- see all four views on the coordinator
+select viewname from pg_views where viewname like 'dependent_view__';
+\c - - - :worker_1_port
+-- see 1st and 3rd view on the worker
+select viewname from pg_views where viewname like 'dependent_view__';
+
+\c - - - :master_port
+
+CREATE TABLE parent_1 (a INT UNIQUE) PARTITION BY RANGE(a);
+SELECT create_distributed_table('parent_1','a');
+CREATE TABLE parent_1_child_1 (a int);
+CREATE TABLE parent_1_child_2 (a int);
+CREATE VIEW v1 AS SELECT * FROM parent_1_child_1;
+CREATE VIEW v2 AS SELECT * FROM parent_1_child_2;
+CREATE VIEW v3 AS SELECT parent_1_child_2.* FROM parent_1_child_2 JOIN parent_1_child_1 USING(a);
+CREATE VIEW v4 AS SELECT * FROM v3;
+
+
+alter table parent_1 attach partition parent_1_child_1 FOR VALUES FROM (0) TO (10) ;
+
+-- only v1 distributed
+SELECT run_command_on_workers($$SELECT count(*) FROM v1$$);
+SELECT run_command_on_workers($$SELECT count(*) FROM v2$$);
+SELECT run_command_on_workers($$SELECT count(*) FROM v3$$);
+SELECT run_command_on_workers($$SELECT count(*) FROM v4$$);
+
+-- all views becomes distributed
+alter table parent_1 attach partition parent_1_child_2 FOR VALUES FROM (10) TO (20);
+
+SELECT run_command_on_workers($$SELECT count(*) FROM v1$$);
+SELECT run_command_on_workers($$SELECT count(*) FROM v2$$);
+SELECT run_command_on_workers($$SELECT count(*) FROM v3$$);
+SELECT run_command_on_workers($$SELECT count(*) FROM v4$$);
+
```

Comment:
Could you please check pg_dist_object as well to show views are marked as distributed?